src/data/resolvers/colony.ts

```diff
@@ -119,38 +125,60 @@ export const getProcessedColony = async (
 
   try {
     if (ipfsMetadata) {
-      const {
-        colonyDisplayName = null,
-        colonyAvatarHash = null,
-        colonyTokens = [],
-        verifiedAddresses = [],
-        isWhitelistActivated = null,
-      } = JSON.parse(ipfsMetadata);
-
-      displayName = colonyDisplayName;
-      avatarHash = colonyAvatarHash;
-      tokenAddresses = colonyTokens;
-      whitelistedAddresses = verifiedAddresses;
-      if (isWhitelistActivated !== null) {
-        whitelistActivated = isWhitelistActivated;
+      const metadataVersion = getEventMetadataVersion(ipfsMetadata);
+      if (metadataVersion === 1) {
+        /*
+         * original metadata format
+         */
+        const {
+          colonyDisplayName = null,
+          colonyAvatarHash = null,
+          colonyTokens = [],
+          verifiedAddresses = [],
+          isWhitelistActivated = null,
+        } = JSON.parse(ipfsMetadata);
+
+        displayName = colonyDisplayName;
+        avatarHash = colonyAvatarHash;
+        tokenAddresses = colonyTokens;
+        whitelistedAddresses = verifiedAddresses;
+        if (isWhitelistActivated !== null) {
+          whitelistActivated = isWhitelistActivated;
+        }
+      } else {
+        /*
+         * new metadata format
+         */
+        const colonyMetadata = getColonyMetadataFromResponse(ipfsMetadata);
+        displayName = colonyMetadata?.colonyDisplayName || '';
+        avatarHash = colonyMetadata?.colonyAvatarHash || '';
+        tokenAddresses = colonyMetadata?.colonyTokens || [];
+        whitelistedAddresses = colonyMetadata?.verifiedAddresses || [];
+        if (colonyMetadata?.isWhitelistActivated) {
+          whitelistActivated = colonyMetadata.isWhitelistActivated;
+        }
```

Comment:
The dapp needs to account for cases where the IPFS data is set to something we don't expect _(number, empty string, empty object, anything else really)_ since that can be set directly, and not through the dapp.

So in this case `getEventMetadataVersion` will return `1` since it can't parse, but our `JSON.parse` I think will fail for something that's not a string. 

I can't remember exactly how the return value looks like, so this might actually not even be a problem, but please investing nonetheless

Reply:
So I tried not to touch the existing logic that handles Metadata.

When trying to ascertain the MetadataVersion, just as what currently happens, with V1 we make a best effort by assuming thats its V1 (not version tagged), and then assuming the data is consistent & coherent.

With V2 things are improved by using schema validation (Yup) which protects against (most of) the unexpected. So the data should conform and be consistent & coherent.

If i follow your example:
 1. Some 'gobbledegook' is found at the IPFS hash
 2. Try to get the version number, which fails, so V1 is assumed (we have no other choice)
 3. V1 parsing logic is performed, which is what currently happens via JSON.parse in the dapp
 4. Data for property may or may not be found/parsed, so either property has value or null

If we are close to moving to V2 i don't think it is worth the effort to implement schema validation on V1 data or adding logic to ColonyEventMetadataParser to protect V1 dapp logic.